platform/broadcom/sonic-platform-modules-dell/s6000/sonic_platform/sfp.py

```diff
@@ -437,12 +458,69 @@ def get_presence(self):
         # Mask off the bit corresponding to our port
         mask = (1 << self.sfp_ctrl_idx)
 
+        fcntl.flock(fd, fcntl.LOCK_UN)
+
         # ModPrsL is active low
         if ((reg_value & mask) == 0):
             return True
 
         return False
 
+    def get_modsel(self):
+        modsel_ctrl = self.sfp_control + 'qsfp_modsel'
+        try:
+            reg_file = open(modsel_ctrl, "r+")
+        except IOError as e:
+            return False
+
+        reg_hex = reg_file.readline().rstrip()
+
+        # content is a string containing the hex
+        # representation of the register
+        reg_value = int(reg_hex, 16)
+
+        # Mask off the bit corresponding to our port
+        index = self.sfp_ctrl_idx
+
+        mask = (1 << index)
+
+        if ((reg_value & mask) == 1):
+            modsel_state = False
+        else:
+            modsel_state = True
+
+        return modsel_state
+
+    def set_modsel(self):
```

Comment:
If you bring the lock inside set_modsel(), then you don't have to lock/unlock every place where set_modsel() is called, keeps code changes to minimum.

Reply:
The lock is based on set_modsel() but we cant introduce it here.
A scenario:
1) When sfputil tries to read a port, it checks for modprs register and to read that we are setting modsel register.
2) Before we actually reading modprs register, if pmon docker or any other process trying to set modsel register then it will have the race condition here.
3) To avoid this we are locking before modsel register is being set and releasing the lock only after the requirements are over(In our case after retriving modprs register).